File: 443-string-compression/string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int j = 0, i = 0;
        while (i < chars.size()) {
            int count = 1;
            while (i + count < chars.size() && chars[i] == chars[i + count])
                count++;
            chars[j++] = chars[i];

            if (count > 1) {
                for (auto c : to_string(count)) 
                    chars[j++] = c;
            }
            i += count;
        }
        return j;
    }
};

File: 443-string-compression/string-compression.go
func compress(chars []byte) int {
    l := 0
    ch := chars[0]
    count := 0

    for r := 0; r <= len(chars); r++ {
        if r < len(chars) && ch == chars[r] {
            count++
        } else {
            chars[l] = ch
            l++
            if count > 1 {
                num := strconv.Itoa(count)
                for i := 0; i < len(num); i, l = i + 1, l + 1 {
                    chars[l] = num[i]
                }
            }
            if r < len(chars) {
                ch = chars[r]
                count = 1
            }
        }
    }

   return l
}